zookeeper-server/src/main/java/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -155,13 +155,11 @@
 
     private final int sessionTimeout;
 
-    private final ZooKeeper zooKeeper;
-
-    private final ClientWatchManager watcher;
+    private final ZKWatchManager watchManager;
```

Comment:
this one should be ClientWatchManager

Reply:
Nope *for now*. `ClientWatchManager` provides the only interface `materialize` while we need to retrieve dataWatches and so on. It can be a separated issue we revisit `ClientWatchManager` and enrich interfaces it provides, as well as loose this type constrain to `ClientWatchManager`.